src/logbooks/logbooks.service.ts

```diff
@@ -132,7 +144,10 @@ export class LogbooksService {
     if (this.logbookEnabled) {
       if (this.username && this.password) {
         try {
-          const accessToken = await this.login(this.username, this.password);
+          const shouldRenewToken = await this.shouldRenewToken();
+          if (!this.accessToken || shouldRenewToken) {
```

Comment:
Question: Do you think checking for if `this.accessToken` is falsy could be done in the `shouldRenewToken()` instead?

Reply:
That makes code much cleaner, I have changed them in the new commit. Thanks!